### PR TITLE
Remove deprecated `on_{save,load}_checkpoint` signature

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

See title

### Does your PR introduce any breaking changes? If yes, please list them.

Removes deprecated code

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/core/saving.py [hunk 1]
```diff
@@ -212,25 +212,6 @@ def _load_model_state(cls, checkpoint: Dict[str, Any], strict: bool = True, **cl
...
-    def on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
-        """
-        Give the model a chance to add something to the checkpoint.
-        ``state_dict`` is already there.
-
-        Args:
-            checkpoint: A dictionary in which you can save variables to save in a checkpoint.
-                Contents need to be pickleable.
-        """
-
```
Comment (carmocca) [inline pytorch_lightning/core/saving.py:233]: This was duplicated in the `CheckpointHooks` mixin

**Review by awaelchli — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -77,6 +77,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed the deprecated `optimizer_idx` from `training_step` as an accepted argument in manual optimization ([#8576](https://github.com/PyTorchLightning/pytorch-lightning/pull/8576))
 
 
+- Removed support for the deprecated `on_save_checkpoint` signature. The hook now takes a `checkpoint` positional parameter ([#8697](https://github.com/PyTorchLightning/pytorch-lightning/pull/8697))
+
+
+- Removed support for the deprecated `on_load_checkpoint` signature. The hook now takes a `model` positional parameter ([#8697](https://github.com/PyTorchLightning/pytorch-lightning/pull/8697))
```
Comment (yifuwang) [inline CHANGELOG.md:83]: nit
```suggestion
- Removed support for the deprecated `on_load_checkpoint` signature. The hook now takes a `pl_module` positional parameter ([#8697](https://github.com/PyTorchLightning/pytorch-lightning/pull/8697))
```

**Review by Borda — APPROVED** [review]

File: tests/deprecated_api/test_remove_1-5.py [hunk 3]
```diff
@@ -30,104 +29,6 @@
...
-    trainer.fit(model)
-
-    with pytest.deprecated_call(match="old signature will be removed in v1.5"):
-        trainer_kwargs["max_steps"] = 2
-        cb = OldSignatureOnLoadCheckpoint()
-        trainer = Trainer(**trainer_kwargs, callbacks=cb, resume_from_checkpoint=chk.last_model_path)
-        trainer.fit(model)
-        assert cb.on_load_checkpoint_called
-
-    class ValidSignature1(BaseSignatureOnLoadCheckpoint):
```
Comment (tchaton) [inline tests/deprecated_api/test_remove_1-5.py:112]: Should we keep the valid signature verification and move this to callback ?

Comment (carmocca) [reply tests/deprecated_api/test_remove_1-5.py:112]: What for? We don't do it for other hooks, and the deprecated calls were removed

**Review by ananthsub — APPROVED** [review]
